lib/beaker/hypervisor/docker.rb

```diff
@@ -21,6 +21,15 @@ def initialize(hosts, options)
 
       # Pass on all the logging from docker-api to the beaker logger instance
       ::Docker.logger = @logger
+
+      if ::Docker.version['Version'] =~ /swarm/
+        @docker_type = 'swarm'
+        raise "Using Swarm with beaker requires a private regitry. Please setup the private registry and set the 'DOCKER_REGISTRY' env var" unless ENV['DOCKER_REGISTRY']
```

Comment:
Structured like this;

```
unless ENV['DOCKER_REGISTRY']
  raise "Using Swarm with beaker requires a private regitry. Please setup the private registry and set the 'DOCKER_REGISTRY' env var"
end
```

You can see the condition, and it really is just a long string.